### PR TITLE
[WIP/Draft] Mistral/Mixtral inference

### PR DESCRIPTION
**Pull request by WendyShang**

### CODE REVIEW

File: MaxText/layers/llama2.py [hunk 1]
```diff
@@ -116,18 +118,50 @@ def __call__(self,
...
+            #batch_idx, nth_expert = jnp.where(selected_experts == k)
+            weights_exp = jnp.sum(jnp.multiply(selected_experts==k, weights), axis=-1)
+            mlp_lnx_exp = linears.MlpBlock(
+            intermediate_dim=cfg.mlp_dim,
+            activations=cfg.mlp_activations,
+            intermediate_dropout_rate=cfg.dropout_rate,
+            dtype=cfg.dtype,
+            name=f'mlp_{k}',
+            config=cfg,
+            )(hidden_states, deterministic=deterministic)
```
Comment (WendyShang) [inline MaxText/layers/llama2.py:146]: it may also be slightly faster if we mask hidden_states. i havent tested

**Comment by michelle-yooh** [comment]
pytype check for this particular PR was running indefinitely. I had to cancel them to unblock the queue. Please investigate.

File: MaxText/layers/llama2.py [hunk 2]
```diff
@@ -116,18 +118,50 @@ def __call__(self,
...
-        intermediate_dropout_rate=cfg.dropout_rate,
-        dtype=cfg.dtype,
-        name='mlp',
-        config=cfg,
-    )(hidden_states, deterministic=deterministic)
-    mlp_lnx = nn.with_logical_constraint(
-        mlp_lnx, ('activation_batch', 'activation_length', 'activation_embed')
-    )
+    if cfg.num_experts > 1:
+        gate_logits = linears.DenseGeneral(
```
[the remark this answers is not in the document]
Comment (RissyRan) [reply MaxText/layers/llama2.py:122]: sure thing!

File: MaxText/pyconfig.py [hunk 3]
```diff
@@ -51,7 +51,7 @@ def validate_attention_type(s: str) -> bool:
     )
 
 def validate_model_name(s: str) -> bool:
-  valid_model_names= ('default', 'llama2-7b') # currently supported models
+  valid_model_names= ('default', 'llama2-7b', 'llama2-8x7b') # currently supported models
```
[the remark this answers is not in the document]
Comment (RissyRan) [reply MaxText/pyconfig.py:54]: Sure, I think mistral-8x7b should be name, please see [here](https://screenshot.googleplex.com/C4Y3xEsVAd3RKDZ).
